dojo/tools/qualys_infrascan_webgui/parser.py

```diff
@@ -66,146 +40,78 @@ def issue_r(raw_row, vuln, scan_date):
-        sev = None
-        if _temp['CVSS_score'] is not None and float(_temp['CVSS_score']) > 0:
-            if 0.1 <= float(_temp['CVSS_score']) <= 3.9:
-                sev = 'Low'
-            elif 4.0 <= float(_temp['CVSS_score']) <= 6.9:
-                sev = 'Medium'
-            elif 7.0 <= float(_temp['CVSS_score']) <= 8.9:
-                sev = 'High'
-            elif float(_temp['CVSS_score']) >= 9.0:
-                sev = 'Critical'
-        elif vuln_details.get('severity') is not None:
-            if int(vuln_details.get('severity')) == 1:
-                sev = 'Informational'
-            elif int(vuln_details.get('severity')) == 2:
-                sev = 'Low'
-            elif int(vuln_details.get('severity')) == 3:
-                sev = 'Medium'
-            elif int(vuln_details.get('severity')) == 4:
-                sev = 'High'
-            elif int(vuln_details.get('severity')) == 5:
-                sev = 'Critical'
-        elif sev is None:
-            sev = 'Informational'
-        finding = None
-        if _temp_cve_details:
-            refs = "\n".join(list(_cl.values()))
-            finding = Finding(title=_temp['vuln_name'],
-                              mitigation=_temp['solution'],
-                              description=_temp['vuln_description'],
-                              severity=sev,
-                              references=refs,
-                              impact=_temp['IMPACT'],
-                              )
+    for vuln_cat in raw_row.findall('VULNS/CAT'):
+        _category = str(vuln_cat.get('value'))
+        for vuln_details in vuln_cat.findall('VULN'):
+            _temp = issue_row
+
+            _gid = vuln_details.get('number')
+
+            _temp['port_status'] = _port
+
+            _result = str(vuln_details.findtext('RESULT'))
+
+            # Vuln name
+            _temp['vuln_name'] = vuln_details.findtext('TITLE')
+
+            # Vuln Description
+            _description = str(vuln_details.findtext('DIAGNOSIS'))
+            # Solution Strips Heading Workaround(s)
+            _temp['solution'] = htmltext(str(vuln_details.findtext('SOLUTION')))
+
+            # Vuln_description
+            _temp['vuln_description'] = "\n".join([htmltext(_description),
+                                                    htmltext("**Category:** " + _category),
+                                                    htmltext("**QID:** " + str(_gid)),
+                                                    htmltext("**Port:** " + str(_port)),
+                                                    htmltext("**Result Evidence:** " + _result),
+                                                   ])
+            # Impact description
+            _temp['IMPACT'] = htmltext(str(vuln_details.findtext('CONSEQUENCE')))
+
+            # CVE and LINKS
+            _cl = []
+            _temp_cve_details = vuln_details.iterfind('CVE_ID_LIST/CVE_ID')
+            if _temp_cve_details:
```

Comment:
@damiencarol isn't this aggregating several CVE into a single defectdojo finding? I don't know these reports but maybe it makes sense to count the CVE and put that number into nb_occurences. 

If not, I believe it's better to leave nb_occurences to null rather than always hardcoded to 1

Reply:
interesting comment.
FYI all the parsers implements `nb_occurences` this way:
1) always set it to 1 when creating the finding **or** setting to the same value if the source aggregate the things
2) if there is de-duplication in the parser itself, update `nb_occurences` 
To be honest I was wondering to work on updating every parsers to follow these guidelines.
So when you create a single finding and set `nb_occurences` to 1, it's not *hard-coded*, it's the correct value.

I think you're right to point to the pb of this parser that it aggregates all CVE in one finding **but** it's not how the data model works right now.
The parser should create one finding by CVE and copy all other data, as for now `cve` is an attribute of the finding.
I can change that in a future pull request but I don't want this pull request to become a monster PR that change everything in this parser.
@ptrovatelli could we post-pone this big change in another pull request and merge this one as-is?

Reply:
Also if we follow your way, we will have one finding with one CVE that have `nb_occurences` > 1. it could make think the user that the CVE appear more than one time. Which is not the case.

Reply:
initially nb_occurences was meant for parsers that aggregate several occurences of the same vulnerability from the source into a single finding. for example same cve at multiple line number in the same file. The goal of nb_occurences is that when looking at the source tool, ppl can make sure that defectdojo didn't loose finding during the process: i have 10 vuln in my source tool, only one in defectdojo, but as nb_occurences=10 i'm good. that's the case for checkmarx parser for example which tends to count many vulns that are aggretated to a single finding in defectdojo. 

it seems that qualys aggregates cve1, cve2, cve3 into the same finding, so it seems reasonnable to use nb_occurences too, even though it's a little bit different than previous use case. 

anyways if the parser is always setting 1 i think it's better to leave empty. same discussion as "no mitgation found" vs null :)

Reply:
completely agree with your comment.
I think I will work on a PR that dissociate the many CVE into multiple findings to stick to this model.
Could be good also to explain this details in the documentation that help implementing parsers.